integration/tsic/tsic.go

```diff
@@ -175,6 +188,18 @@ func WithNetfilter(state string) Option {
 	}
 }
 
+// WithBuildTag adds an additional value to the `-tags=` parameter
+// of the Go compiler, allowing callers to customize the Tailscale client build.
+// This option is only meaningful when invoked on **HEAD** versions of the client.
+// Attempts to use it with any other version is a bug in the calling code.
+func WithBuildTag(tag string) Option {
+	return func(tsic *TailscaleInContainer) {
```

Comment:
I propose panicking here if the tsic passed is not `HEAD`, just to fail fast so the dev can fix it without faffing around.

Reply:
Done, for now I have opted to panic with the same error value I return from the other checks - although at this point I think it might be safe to remove the build tags check [here](https://github.com/juanfont/headscale/pull/2247/files#diff-2ba815dc33efc14a3e231f81a0cba347b8db54804f18602e12d6410865ffab74R284), but since I wrote it to be generic and check for *any* build time config, it may still catch mistakes on future build options. I am not against removing it if desired, though.